### ISSUE TITLE
Armor not saving insulation values, but still has the material

### ISSUE THREAD

**Issue by Zanklev** [Momo-Studios/Cold-Sweat-Rework-Forge#282]
Really weird issue, updated from version A to C on my server.

At first it was fine but after a restart insulation vanished from our armor, but we can still remove the insulating materials from our armor. However newly applied material will no longer provider insulation. Tested on numerous unmodified armors, all vanilla armors. Armor that has built in insulation (Leather) are fine.

Did a quick test on a fresh SP world and had no issue applying materials, removing, leaving and rejoining. Issue didn't arise.
Possibly an issue with updating the mod on an existing world?

Notice you can see the armor has 2 NBT tags on it despite seemingly no insulation, this armor had been previous insulated with white wool.

*edit* Forgot to mention, I rolled back to version A. I had to add the wool back to my armor but it seems to be working properly again.
https://github.com/Momo-Softworks/Cold-Sweat/assets/139833714/37144855-39e0-406c-b032-191113ef19d4

**Comment by Zanklev** [comment]
New developments. I rebuilt my server from fresh and first thing I did was test version C again. The issue occured again, I tried adding wool to iron armor. It doesn't show the insulation value being added to armor, however it does show a second NBT tag being added to the armor. I'm unsure of whether it's actually added insulation or not.

Once the insulation is added to the iron chest plate, it cannot be removed with shears. It looks like it can be though, but once you take the wool out of the result box it doesn't give you anything.

[latest.log](https://github.com/Momo-Softworks/Cold-Sweat/files/15228467/latest.log)

If it helps, here is the latest.log from my attempt. You can see near the bottom when I tried added the insulation.

**Comment by Zanklev** [comment]
I just tested SP and it works correct there, so this must be a server issue.

[latest.log](https://github.com/Momo-Softworks/Cold-Sweat/files/15228482/latest.log)

**Comment by MikulDev** [comment]
If you open the creative inventory menu on a dedicated server, all mod-specific item data is immediately erased. This is unfortunately a side-effect of how the creative inventory works. 

Forge hasn't made any efforts to fix this issue over the last several years since it was brought to their attention, so I doubt it will be changing anytime soon.

**Comment by Zanklev** [comment]
When using it on an older version of the mod it's not a problem at all, it's odd that it would only be affecting the latest release. The problem was affecting everyone on the server, so if one person goes into creative it ruins everyones armor insulation?

**Comment by MikulDev** [comment]
I'm not sure why it would be working for any build past 2.2, as it's been using fundamentally the same system since then.
And no, opening the creative inventory only deletes the data for the existing items in your inventory. Nobody else is affected, and new items will be fine as well

**Comment by Zanklev** [comment]
Everyone's insulation was vanishing from their armor but the "NBT" tags remained at two as if the insulation was still there, just not showing the insulation bars. Then the behavior of adding and removing insulation is completely wack. This occurred without me having gone into creative mode at all. I didn't go into creative until after the problem occurred because I thought that updating had just messed with some armors so I was 'refunding' the cost by rebuilding the armors but then I noticed the problem continued.

Is this any way I can help narrow this down?

**Comment by Zanklev** [comment]
I just created a test server on my own computer and installed my modpack along with the latest version of Cold Sweat, I did a short playtest, not even granting myself OP on the server to make sure creative mode isn't touched. The issue is still persisting, this definitely seems to be an issue with servers.

[latest.log](https://github.com/Momo-Softworks/Cold-Sweat/files/15245322/latest.log)

**Comment by Zanklev** [comment]
Oh new discovery! It seems to be an issue with wool. I created a fresh unmodded forge server and joined it to test with just Cold Sweat alone to make sure there was no conflictions and the issue still persisted, however I had failed to test the other insulation types before. It seems to be only wool that is bugged.

*edit* I just went in and tested all the insulation materials. It's exclusively the wool that's bugged.

https://www.youtube.com/watch?v=0jkluCUfMS8

Here is a video of me testing the different cold insulation types on the fresh server.

**Comment by MikulDev** [comment]
I think I've found the solution to your problem. There is a bug that sometimes prevents settings from syncing from the dedicated server to the client. I was able to replicate your issue by deleting wool from the configs on the client side, while keeping it on the server. This means that the armor actually is being insulated, but it doesn't look like it because the client doesn't know that wool is an insulation item and thus won't display it on the tooltip.

I've already found and implemented a fix, alongside some other improvements regarding the insulation tooltip, so this will be solved in the next update.
